packages/common-common/src/permissions.ts

```diff
@@ -0,0 +1,56 @@
+export enum Action {
+  INVITE_MEMBERS = 0,
+  BAN_MEMBERS = 1,
+  MANAGE_COMMUNITY = 2,
+  ADMINISTRATOR = 3,
+  MANAGE_ROLES = 4,
+  MANAGE_WEBHOOKS = 5,
+  MANAGE_TOPICS = 6,
+  MANAGE_CHAT_CHANNELS = 7,
+  VIEW_COMMUNITY_INSIGHTS = 8,
+  MANAGE_INVITES = 9,
+  VIEW_TOPIC = 10,
+  VIEW_CHAT_CHANNELS = 11,
+  CREATE_THREAD = 12,
+  MANAGE_THREADS = 13,
+  CREATE_CHAT = 14,
+  CREATE_REACTION = 15,
+  CREATE_COMMENT = 16,
+  CREATE_POLL = 17,
+  VOTE_ON_POLLS = 18,
+  MANAGE_POLLS = 19,
+}
+
+export type Permissions = bigint;
+
+export function isPermitted(permission: Permissions, action: Action): boolean {
+  const actionAsBigInt: bigint = BigInt(1 << action);
```

Comment:
This will fail, e.g. if `action` is very large, because it's doing the operation on a number and then converting to BigInt.

Reply:
I see!